File: code-samples/python/fibonacci.py
#!/usr/bin/python3

def fib1(n):
  a, b = 0, 1
  while a < n:
    print(a, end=' ')
    a, b = b, a+b
  print()


def fib2(n):
  res = []
  a, b = 0, 1
  while a < n:
    res.append(a)
    a, b = b, a+b
  return res

# print fibonacci numbers upto 2000
fib1(2000)

# return a list of fibonacci number upto 2000
f = fib2(2000)
print(f)

File: code-samples/php/fibonacci.php
#!/usr/bin/php
<?php

function fib1($n)
{
  $fib     = array();
  $fib[]   = 0;  # first
  $fib[]   = 1;  # second
  $e       = 1;  # last used element in the array
  # while the value in current element is less than needed value
  while(true)
  {
    $nxtv  = ($fib[$e] + $fib[$e-1]);
    if ($nxtv <= $n)  { $fib[] = $nxtv; $e++; continue; }
    break;
  }
  return $fib;
}

# print fibonacci upto given number
foreach(fib1(2000) as $v) { echo "$v "; }
echo "\n";

# obtain an array of fibonacci upto given number
$f = fib1(2000);
print_r($f);
?>
